src/blueprints/tv2/action-factories/tv2-action-factory-provider.ts

```diff
@@ -0,0 +1,328 @@
+      configuration
+    )
+
+    return this.videoClipActionFactoryInstance.factory
+  }
+
+  public createVideoMixerActionFactory(configuration: Tv2BlueprintConfiguration): Tv2VideoMixerConfigurationActionFactory {
+    this.videoMixerConfigurationActionFactoryInstance = this.getUpdatedActionFactoryInstance(
+      this.videoMixerConfigurationActionFactoryInstance,
+      () => {
+        return new Tv2VideoMixerConfigurationActionFactory(
+          this.createVideoMixerTimelineObjectFactory(configuration)
+        )
+      },
+      (c: Tv2BlueprintConfiguration): boolean => {
+        return this.didVideoMixerTypeChange(c, configuration)
+      },
+      configuration
+    )
+
+    return this.videoMixerConfigurationActionFactoryInstance.factory
+  }
+
+  public createSplitScreenActionFactory(configuration: Tv2BlueprintConfiguration): Tv2SplitScreenActionFactory {
+    this.splitScreenActionFactoryInstance = this.getUpdatedActionFactoryInstance(
+      this.splitScreenActionFactoryInstance,
+      () => {
+        return new Tv2SplitScreenActionFactory(
+          this.createActionManifestMapper(),
+          this.createVideoMixerTimelineObjectFactory(configuration),
+          this.createAudioTimelineObjectFactory(),
+          this.createGraphicsSplitScreenTimelineObjectFactory(),
+          this.createVideoClipTimelineObjectFactory(),
+          this.createAssetPathHelper()
+        )
+      },
+      (c: Tv2BlueprintConfiguration): boolean => {
+        return this.didVideoMixerTypeChange(c, configuration)
+      },
+      configuration
+    )
+
+    return this.splitScreenActionFactoryInstance.factory
+  }
+
+  private createGraphicsSplitScreenTimelineObjectFactory(): Tv2GraphicsSplitScreenTimelineObjectFactory {
+    return new Tv2CasparCgTimelineObjectFactory(this.createAssetPathHelper())
+  }
+
+  public createReplayActionFactory(configuration: Tv2BlueprintConfiguration): Tv2ReplayActionFactory {
+    this.replayActionFactoryInstance = this.getUpdatedActionFactoryInstance(
+      this.replayActionFactoryInstance,
+      () => {
+        return new Tv2ReplayActionFactory(
+          this.createVideoMixerTimelineObjectFactory(configuration),
+          this.createAudioTimelineObjectFactory()
+        )
+      },
+      (c: Tv2BlueprintConfiguration): boolean => {
+        return this.didVideoMixerTypeChange(c, configuration)
+      },
+      configuration
+    )
+
+    return this.replayActionFactoryInstance.factory
+  }
+
+  public createRobotActionFactory(configuration: Tv2BlueprintConfiguration): Tv2RobotActionFactory {
+    this.robotActionFactoryInstance = this.getUpdatedActionFactoryInstance(
+      this.robotActionFactoryInstance,
+      () => {
+        return new Tv2RobotActionFactory(
+          this.createRobotTimelineObjectFactory()
+        )
+      },
+      (): boolean => {
+        return false // Currently we have no configuration changes that warrant a new RobotActionFactory.
+      },
+      configuration
+    )
```

Comment:
Personal opinion incoming: When we have lambda functions that have a simple expression that is returned right away, the curly brackets and `return` is unnecessary. For me it becomes more cluttered to read, e.g. since it leads to more indentation in the code. I find the below code a bit easier to read :)
```typescript
    this.robotActionFactoryInstance = this.getUpdatedActionFactoryInstance(
      this.robotActionFactoryInstance,
      () => new Tv2RobotActionFactory(this.createRobotTimelineObjectFactory()),
      (): boolean => false, // Currently we have no configuration changes that warrant a new RobotActionFactory.
      configuration
    )
```